packages/web3/types.d.ts

```diff
@@ -0,0 +1,467 @@
+  }
+  currentProvider: Provider
+  estimateGas(tx: Tx, callback?: Callback<number>): Promise<number>
+  getAccounts(cb?: Callback<Array<string>>): Promise<Array<string>>
+  getBalance(address: string, defaultBlock?: BlockType, cb?: Callback<number>): Promise<number>
+  getBlock(number: BlockType, returnTransactionObjects?: boolean, cb?: Callback<Block>): Promise<Block>
+  getBlockNumber(callback?: Callback<number>): Promise<number>
+  getBlockTransactionCount(number: BlockType | string, cb?: Callback<number>): Promise<number>
+  getBlockUncleCount(number: BlockType | string, cb?: Callback<number>): Promise<number>
+  getCode(address: string, defaultBlock?: BlockType, cb?: Callback<string>): Promise<string>
+  getCoinbase(cb?: Callback<string>): Promise<string>
+  getCompilers(cb?: Callback<string[]>): Promise<string[]>
+  getGasPrice(cb?: Callback<number>): Promise<number>
+  getHashrate(cb?: Callback<number>): Promise<number>
+  getPastLogs(options: {
+    fromBlock?: BlockType
+    toBlock?: BlockType
+    address: string
+    topics?: Array<string | Array<string>>
+  }, cb?: Callback<Array<Log>>): Promise<Array<Log>>
+  getProtocolVersion(cb?: Callback<string>): Promise<string>
+  getStorageAt(address: string, defaultBlock?: BlockType, cb?: Callback<string>): Promise<string>
+  getTransactionReceipt(hash: string, cb?: Callback<TransactionReceipt>): Promise<TransactionReceipt>
+  getTransaction(hash: string, cb?: Callback<Transaction>): Promise<Transaction>
+  getTransactionCount(address: string, defaultBlock?: BlockType, cb?: Callback<number>): Promise<number>
+  getTransactionFromBlock(block: BlockType, index: number, cb?: Callback<Transaction>): Promise<Transaction>
+  getUncle(blockHashOrBlockNumber: BlockType | string, uncleIndex: number, returnTransactionObjects?: boolean, cb?: Callback<Block>): Promise<Block>
+  getWork(cb?: Callback<Array<string>>): Promise<Array<string>>
+  givenProvider: Provider
+  isMining(cb?: Callback<boolean>): Promise<boolean>
+  isSyncing(cb?: Callback<boolean>): Promise<boolean>
+  net: Net
+  personal: Personal
+  signTransaction(tx: Tx, address?: string, cb?: Callback<string>): Promise<EncodedTransaction>
+  sendSignedTransaction(data: string, cb?: Callback<string>): PromiEvent<TransactionReceipt>
+  sendTransaction(tx: Tx, cb?: Callback<string>): PromiEvent<TransactionReceipt>
+  submitWork(nonce: string, powHash: string, digest: string, cb?: Callback<boolean>): Promise<boolean>
+  sign(address: string, dataToSign: string, cb?: Callback<string>): Promise<string>
+
+
+}
+export declare class Net {
+  getId(cb?: Callback<number>): Promise<number>
+  isListening(cb?: Callback<boolean>): Promise<boolean>
+  getPeerCount(cb?: Callback<number>): Promise<number>
+}
+export declare class Personal {
+  newAccount(password: string, cb?: Callback<boolean>): Promise<boolean>
+  getAccounts(cb?: Callback<Array<string>>): Promise<Array<string>>
+  importRawKey()
+  lockAccount()
+  unlockAccount()
+  sign()
```

Comment:
The above 4 lines should have annotated return types (`: void`).

(Sorry for the drive-by comment, just tried to build this branch and got some errors with the latest TypeScript in strict mode. There were others too.)